wandb/sdk/launch/agent/agent.py

```diff
@@ -236,12 +284,16 @@ def loop(self) -> None:
                 for thread_id in self.thread_ids:
                     self._update_finished(thread_id)
                 if self._ticks % 2 == 0:
-                    if len(self._jobs) == 0:
+                    if len(self.thread_ids) == 0:
                         self.update_status(AGENT_POLLING)
                     else:
                         self.update_status(AGENT_RUNNING)
                     self.print_status()
-                time.sleep(AGENT_POLLING_INTERVAL)
+
+                if self.num_running_schedulers == 0:
```

Comment:
|| `self.num_running_jobs == self.max_jobs` maybe? I hesitate to always poll on 1 second intervals if there's schedulers